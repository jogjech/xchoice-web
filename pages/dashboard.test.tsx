import React from "react";
import { shallow, ShallowWrapper } from "enzyme";
import Dashboard from "./dashboard";

describe("Dashboard page", () => {
  let wrapper: ShallowWrapper;

  beforeEach(() => {
    wrapper = shallow(<Dashboard></Dashboard>);
  });

  it("renders correctly", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("contains layout", () => {
    const layout = wrapper.find("Layout");
    expect(layout.exists()).toBeTruthy();
    expect(layout.prop("title")).toBe("My Surveys");
  });

  it("contains loading survey card", () => {
    const surveyCard = wrapper.find("SurveyCard");
    expect(surveyCard.exists()).toBeTruthy();
    expect(surveyCard.prop("surveyMetadata")).toBe(null);
  });
});
