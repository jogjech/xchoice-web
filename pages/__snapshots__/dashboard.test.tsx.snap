// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard page renders correctly 1`] = `
<div>
  <Layout
    title="My Surveys"
  >
    <div>
      <p
        className="title"
      >
        My Surveys
      </p>
    </div>
    <div>
      <SurveyCard
        surveyMetadata={null}
      />
    </div>
  </Layout>
</div>
`;
